storage/pebble/transactions.go

```diff
@@ -42,9 +46,23 @@ func (t *Transactions) Get(ID common.Hash) (*gethTypes.Transaction, error) {
 		return nil, err
 	}
 
-	tx := &gethTypes.Transaction{}
-	if err := tx.UnmarshalBinary(val); err != nil {
-		return nil, err
+	var evmTxData models.FlowEVMTxData
+
+	if val[0] == types.DirectCallTxType {
+		directCall, err := types.DirectCallFromEncoded(val)
+		if err != nil {
+			return nil, fmt.Errorf("failed to rlp decode direct call: %w", err)
+		}
+
+		evmTxData = models.DirectCallTx{Call: directCall}
+	} else {
+		tx := &gethTypes.Transaction{}
+		if err := tx.UnmarshalBinary(val[1:]); err != nil {
+			return nil, fmt.Errorf("failed to rlp decode transaction: %w", err)
+		}
+
+		evmTxData = models.GethTx{Tx: tx}
 	}
```

Comment:
I would extract this into the transaction model, add a method there which is something like: `UnmarshalTransaction` and then it returns the `Transaction` interface.